src/Directory.cpp

```diff
@@ -0,0 +1,35 @@
+#include "Directory.h"
+
+string Directory::m_previousDir;
+
+string Directory::getCurrentDirectory() {
+    char currentDir[256];
+    if (getcwd(currentDir, sizeof(currentDir)) != nullptr) {
```

Comment:
This way of getting cwd is a bit old fashion, check this out (a more modern way): https://en.cppreference.com/w/cpp/filesystem/current_path